lib/shoulda/matchers/active_model/validate_confirmation_of_matcher.rb

```diff
@@ -94,28 +94,48 @@ def simple_description
 
         def matches?(subject)
           super(subject)
-
-          disallows_different_value &&
-            allows_same_value &&
-            allows_missing_confirmation
+          type = find_type(subject)
```

Comment:
If `find_type` no longer takes a `subject`, then you don't really need to assign a variable to the method -- you can use it as though it were a variable, and that means you don't need to pass it to every place that needs it. What do you think about renaming `find_type` to simply `attribute_type`?